src/tests/unit_tests/tools/test_search_tool.py

```diff
@@ -1,15 +1,87 @@
+
+
+
 from sherpa_ai.config import AgentConfig
 from sherpa_ai.tools import SearchTool
+from sherpa_ai.output_parser import TaskAction
 
 
 def test_search_query_includes_gsite_config():
     site = "https://www.google.com"
     config = AgentConfig(
         verbose=True,
         gsite=site,
+        search_domains = [site],
+        invalid_domain = []
+    )
+    search_tool = SearchTool(config=config)
+
+    query = "What is the weather today?"
+    search_result = search_tool._run(query)
+    assert search_result is not None
+    assert search_result is not ""
```

Comment:
When comparing things other than `None`, we should use `==`. This applies to other places as well.